daemon: Add DRIVER_MODE variable

lah7 opened the pull request:
> As reviewed in #2415

This fixes the BlackWidow V4 Pro being unable to apply a 8000 Hz poll rate when the daemon starts, as the device reset to 1000 Hz after entering "driver mode", defeating the point of persistence.

Adding a variable allows this device to enter driver mode first while retaining existing behaviour for all our current devices.

Fixes #2408

Release note suggestion:

> Fix BlackWidow V4 Pro unable to restore poll rates above 1000 Hz.

Edu4rdSHL reviewed (approved):
I confirm that this resolves the issue. Thank you!

lah7 committed:
daemon: Introduce DRIVER_MODE variable

Introduce a variable to declaratively switch the device to driver mode.

An intended side effect is that we now switch to driver mode earlier,
*before* applying persistence.

This fixes a problem for the BlackWidow V4 Pro, where the keyboard would
reset to 1000 Hz after driver mode, defeating the 8000 Hz applied by
persistence.

Fixes #2408

z3ntu reviewed (approved):
LGTM.

Let's keep this waiting a bit longer until the upcoming bugfix release is out.

This deserves to sit some time in master before getting released, and get at least a bit of testing in via Ubuntu Daily PPA, AUR openrazer-git, etc.
